src/main/clojure/clara/rules/durability.clj

```diff
@@ -322,10 +354,22 @@
    with a MemIdx during serialization."
   nil)
 
+(def ^:dynamic *mem-internal*
```

Comment:
Add a type hint of `^List` here on the var instead of repeating it at each callsite below.
```clj
(def ^:dynamic ^List *mem-internal* ...)
```